x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -173,44 +179,56 @@ export const signalRulesAlertType = ({
 
       logger.debug(buildRuleMessage('[+] Starting Signal Rule execution'));
       logger.debug(buildRuleMessage(`interval: ${interval}`));
-      let wroteStatus = false;
+      let wrotePartialFailureStatus = false;
       await ruleStatusService.goingToRun();
 
       // check if rule has permissions to access given index pattern
       // move this collection of lines into a function in utils
       // so that we can use it in create rules route, bulk, etc.
       try {
-        const inputIndex = await getInputIndex(services, version, index);
-        const privileges = await checkPrivileges(services, inputIndex);
-
-        const indexNames = Object.keys(privileges.index);
-        const [indexesWithReadPrivileges, indexesWithNoReadPrivileges] = partition(
-          indexNames,
-          (indexName) => privileges.index[indexName].read
-        );
+        const hasTimestampOverride = timestampOverride != null && !isEmpty(timestampOverride);
+        const [privileges, timestampFieldCaps] = await Promise.all([
+          pipe(
+            { services, version, index },
+            ({ services: svc, version: ver, index: idx }) =>
+              pipe(
```

Comment:
Ahh, I see: you're using this inner `pipe` to close over `svc` for `checkPrivileges`; this pipe isn't purely a function of the previous outputs. IMO this would probably be cleaner as two imperative function invocations like you had originally, but I would guess that short-circuiting is the main reason for using fp-ts in this file?

I can't find my original comment here, and I think this is acceptable however you want to slice it, but I would suggest keeping the code as self-evident as possible: both `pipe` and `flow` imply a pure pipeline, which isn't what we have here an so any fp-ts usage is going to be somewhat confusing.

Reply:
I was trying to rectify the comment @FrankHassanabad mentioned about pumping outputs of functions into inputs and figured that this might be a good use case for utilizing fp-ts but if you think this is just going to make things more obtuse I'll remove it.

Reply:
this one here https://github.com/elastic/kibana/pull/87489#discussion_r556192534